src/fluree/db/index.cljc

```diff
@@ -355,20 +304,80 @@
   [node]
   (-> node ::expanded true?))
 
+(defn trim-leaf
+  "Remove flakes from the index leaf node `leaf` that are outside of the interval
+  defined by `start-flake` and `end-flake`. nil values for either `start-flake`
+  or `end-flake` makes that side of the interval unlimited."
+  [leaf start-flake end-flake]
+  (cond
+    (and start-flake end-flake)
+    (update leaf :flakes flake/subrange >= start-flake <= end-flake)
+
+    start-flake
+    (update leaf :flakes flake/subrange >= start-flake)
+
+    end-flake
+    (update leaf :flakes flake/subrange <= end-flake)
+
+    :else
+    leaf))
+
+(defn trim-branch
+  "Remove child nodes from the index branch node `branch` that do not contain
+  flakes in the interval defined by `start-flake` and `end-flake`. nil values
+  for either `start-flake` or `end-flake` makes that side of the interval
+  unlimited."
+  [{:keys [children] :as branch} start-flake end-flake]
+  (let [start-key (some->> start-flake (flake/nearest children <=) key)
+        end-key   (some->> end-flake (flake/nearest children <=) key)]
+    (cond
+      (and start-key end-key)
+      (update branch :children flake/subrange >= start-key <= end-key)
+
+      start-key
+      (update branch :children flake/subrange >= start-key)
+
+      end-key
+      (update branch :children flake/subrange <= end-key)
+
+      :else
+      branch)))
+
+(defn trim-node
+  "Remove flakes or children from the index leaf or branch node `node` that are
+  outside of the interval defined by `start-flake` and `end-flake`. nil values
+  for either `start-flake` or `end-flake` makes that side of the interval
+  unlimited."
+  [node start-flake end-flake]
+  (if (leaf? node)
+    (trim-leaf node start-flake end-flake)
+    (trim-branch node start-flake end-flake)))
+
+(defn try-resolve
+  [r start-flake end-flake error-ch node]
+  (go
+    (try* (let [resolved (<? (resolve r node))]
+            (trim-node resolved start-flake end-flake))
+          (catch* e
+                  (log/error e
```

Comment:
Can you indent the body of the `catch*` like `catch`? I.e. two spaces more than `(catch* ...`